packages/server/src/wallet/db/queries/channels.ts

```diff
@@ -64,21 +64,30 @@ async function updateAllocatorChannel(
     holdings?: Uint256;
     id?: number;
     participants?: any[];
+    guaranteedChannel: string;
   }
-  let upserts: Upsert = { channel_id: channelId, commitments, rules_address, nonce };
+  let upserts: Upsert = {
+    channel_id: channelId,
+    commitments,
+    rules_address,
+    nonce,
+    guaranteedChannel,
+  };
 
-  // For now, we just _assume_ that the channel is fully funded
-  const holdings = allocations(hubCommitment)
-    .map(x => x.amount)
-    .reduce((a, b) =>
+  // TODO: We are currently using the allocations to set the funding amount
+  // This assumes that the channel is funded and DOES NOT work for guarantor channels
+  const hubAllocationAmounts = allocations(hubCommitment).map(x => x.amount);
+  let holdings = bigNumberify(0).toHexString();
+  if (hubAllocationAmounts.length > 0) {
+    holdings = hubAllocationAmounts.reduce((a, b) =>
```

Comment:
If you pass the optional initial value to `reduce`, you can make `holdings` a constant, and remove the check on `hubAllocationAmounts.length > 0`

`hubAllocationAmounts.reduce( [callback in L83-86], bigNumberify(0).toHexString())`